src/libs/antares/inifile/inifile.cpp

```diff
@@ -235,44 +153,102 @@ uint IniFile::Section::size() const
     return count;
 }
 
-bool IniFile::open(const AnyString& filename, bool warnings)
+bool startingSection(std::string line)
 {
-    // clear
-    clear();
-    pFilename = filename;
+    boost::trim(line);
+    if (boost::starts_with(line, "[") && boost::ends_with(line, "]"))
+        return true;
+    return false;
+}
+
+std::string getSectionName(std::string line)
+{
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("[]"));
+    return splitLine[1];
+}
+
+bool isProperty(std::string line)
+{
+    if (std::count(line.begin(), line.end(), '=') == 1)
+        return true;
+    return false;
+}
+
+std::pair<std::string, std::string> getKeyValuePair(std::string line)
+{
+    boost::trim(line);
+
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("="));
+
+    return std::make_pair(splitLine[0], splitLine[1]);
+}
+
+bool isComment(std::string line)
+{
+    boost::trim_left(line);
+    if (boost::starts_with(line, "#") || boost::starts_with(line, ";"))
+        return true;
+    return false;
+}
 
-    // Load the file
-    IO::File::Stream file;
-    if (file.open(filename))
+bool isEmpty(std::string line)
+{
+    boost::trim_left(line);
+    return line.empty();
+}
+
+bool IniFile::readStream(std::istream& in_stream)
+{
+    std::string line;
+    IniFile::Section* currentSection(nullptr);
+    while (std::getline(in_stream, line))
     {
-        enum
+        if (startingSection(line))
         {
-            lineHardLimit = 2048
-        };
-        auto* line = new char[lineHardLimit];
+            currentSection = addSection(getSectionName(line));
+            continue;
+        }
 
-        IniFile::Section* lastSection = nullptr;
-        uint64_t read = 0;
+        if (isProperty(line))
+        {
+            std::pair<std::string, std::string> pair = getKeyValuePair(line);
+            currentSection->add(pair.first, pair.second);
+            continue;
+        }
 
-        // analyzing each line
-        while (file.readline(line, lineHardLimit))
-            lastSection = AnalyzeIniLine(pFilename, this, lastSection, line, read);
+        if (isComment(line) || isEmpty(line))
+            continue;
 
-        delete[] line;
+        logs.error() << "Invalid INI line format : '" << line << "'";
+        return false;
+    }
+    return true;
+}
 
-        if (read)
-            Statistics::HasReadFromDisk(read);
+bool IniFile::open(const AnyString& filename, bool warnings)
+{
+    clear();
+    filename_ = filename; // storing filename for further use
+    std::string filePath = filename.to<std::string>();
+
+    if (std::ifstream file(filePath); file.is_open())
+    {
+        if (!readStream(file) && warnings)
+        {
+            logs.error() << "Invalid INI file : " << filePath;
+            return false;
+        }
```

Comment:
This is very error-prone. If `warning=false`, the function will return `true` even though `readStream` failed.

IMHO `warnings` should decide only on logs, the execution flow shouldn't care about `warnings`. Use nested `if` here.

Reply:
ok, you're totally right.
I introduced an extra check to the original code.
The first use of **warnings** is not required. 
Let me explain. In code : 
```C++
    if (std::ifstream file(filePath); file.is_open())
    {
        if (!readStream(file) && warnings) // Here warnings is not used in the previous code
        {
            logs.error() << "Invalid INI file : " << filePath;
            return false;
        }
        return true;
    }

    if (warnings)
    {
        logs.error() << "I/O error: " << filename << ": Impossible to read the file";
    }
```
`if (!readStream(file) && warnings)` should be actually `if (!readStream(file))` in order to keep the same behavior we used to have.